From e5ce8dae10bfe4e3e14e62000a183fd60aacf8f2 Mon Sep 17 00:00:00 2001
From: Meng Mingming <mingming.meng@nxp.com>
Date: Wed, 20 Dec 2017 09:30:27 +0800
Subject: [PATCH 3121/5242] MLK-17204-2 drm/imx: dpu: Set driver data as null
 when to unbind device

commit  1122257b7d486f939405b602e231852b271eddf7 from
https://source.codeaurora.org/external/imx/linux-imx.git

Set driver data as null when to unbind device.

Signed-off-by: Meng Mingming <mingming.meng@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-blit.c |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-blit.c b/drivers/gpu/drm/imx/dpu/dpu-blit.c
index 164f651..fb9aea8 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-blit.c
+++ b/drivers/gpu/drm/imx/dpu/dpu-blit.c
@@ -226,6 +226,7 @@ static void dpu_bliteng_unbind(struct device *dev, struct device *master,
 	list_del(&bliteng->list);
 
 	dpu_bliteng_fini(dpu_bliteng);
+	dev_set_drvdata(dev, NULL);
 
 	imx_dpu_num--;
 
@@ -262,6 +263,9 @@ static int dpu_bliteng_suspend(struct device *dev)
 	struct dpu_bliteng *dpu_bliteng = dev_get_drvdata(dev);
 	int ret;
 
+	if (dpu_bliteng == NULL)
+		return 0;
+
 retry:
 	ret = dpu_be_get(dpu_bliteng);
 	if (ret == -EBUSY)
@@ -280,7 +284,8 @@ static int dpu_bliteng_resume(struct device *dev)
 {
 	struct dpu_bliteng *dpu_bliteng = dev_get_drvdata(dev);
 
-	dpu_bliteng_init(dpu_bliteng);
+	if (dpu_bliteng != NULL)
+		dpu_bliteng_init(dpu_bliteng);
 
 	return 0;
 }
-- 
1.7.9.5

